### PR TITLE
bulk order api extension

### PR DESCRIPTION
**Pull request by vkrashenko**

### CODE REVIEW

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 1]
```diff
@@ -167,6 +167,77 @@ public class ApiCartController : StorefrontControllerBase
 
         }
 
+
```
Comment (tatarincev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: Could we leave only one (bulk) AddItemsToCart method, and remove old which used for single item addition?

Comment (vkrashenko) [reply VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: yes we can. I fix it

**Comment by yecli** [comment]
Rebased on the current dev branch.
We need to adapt vc-theme-default to merge this request. Old one is not compatible with the changes.

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 2]
```diff
@@ -83,19 +83,25 @@ public async Task<ActionResult<int>> GetCartItemsCount()
...
-        public async Task<ActionResult<ShoppingCartItems>> AddItemToCart([FromBody] AddCartItem cartItem)
+        public async Task<ActionResult<ShoppingCartItems>> AddItemToCart([FromBody] AddCartItem[] cartItems)
         {
             EnsureCartExists();
 
             //Need lock to prevent concurrent access to same cart
             using (await AsyncLock.GetLockByKey(WorkContext.CurrentCart.Value.GetCacheKey()).LockAsync())
             {
                 var cartBuilder = await LoadOrCreateCartAsync();
+                string[] productsId = cartItems.Select(x => x.ProductId).Distinct().ToArray();
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: var productIds

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 3]
```diff
@@ -83,19 +83,25 @@ public async Task<ActionResult<int>> GetCartItemsCount()
...
+                int quantity;
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: Remove

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 4]
```diff
@@ -83,19 +83,25 @@ public async Task<ActionResult<int>> GetCartItemsCount()
...
+                var products = await _catalogService.GetProductsAsync(productsId, Model.Catalog.ItemResponseGroup.Inventory | Model.Catalog.ItemResponseGroup.ItemWithPrices);
 
-                var products = await _catalogService.GetProductsAsync(new[] { cartItem.ProductId }, Model.Catalog.ItemResponseGroup.Inventory | Model.Catalog.ItemResponseGroup.ItemWithPrices);
                 if (products != null && products.Any())
                 {
-                    await cartBuilder.AddItemAsync(products.First(), cartItem.Quantity);
+                    foreach (var product in products)
+                    {
+                        quantity = cartItems.Where(x => x.ProductId == product.Id).Sum(x => x.Quantity);
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: var quantity = ...

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 5]
```diff
@@ -137,37 +143,42 @@ public async Task<ActionResult> ChangeCartItemPrice([FromBody] ChangeCartItemPri
...
             EnsureCartExists();
 
             //Need lock to prevent concurrent access to same cart
             using (await AsyncLock.GetLockByKey(WorkContext.CurrentCart.Value.GetCacheKey()).LockAsync())
             {
                 var cartBuilder = await LoadOrCreateCartAsync();
-
-                var lineItem = cartBuilder.Cart.Items.FirstOrDefault(i => i.Id == changeQty.LineItemId);
-                if (lineItem != null)
+                foreach (var changeQty in changeQuantites)
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: changeQuantity

Multi-line blocks of code should be separated with empty line from other code.

File: VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs [hunk 6]
```diff
@@ -137,37 +143,42 @@ public async Task<ActionResult> ChangeCartItemPrice([FromBody] ChangeCartItemPri
...
+        public async Task<ActionResult<ShoppingCartItems>> RemoveCartItem(string[] lineItemIds)
         {
             EnsureCartExists();
 
             //Need lock to prevent concurrent access to same cart
             using (await AsyncLock.GetLockByKey(WorkContext.CurrentCart.Value.GetCacheKey()).LockAsync())
             {
                 var cartBuilder = await LoadOrCreateCartAsync();
-                await cartBuilder.RemoveItemAsync(lineItemId);
+                foreach (var itemId in lineItemIds)
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs]: lineItemId

Multi-line blocks of code should be separated with empty line from other code.

**Comment by yecli** [comment]
Seems not needed

**Pull request closed without merging**